jit-spell.el

```diff
@@ -529,6 +529,50 @@ With a numeric ARG, move backwards that many misspellings."
            (jit-spell--accept-word (or (match-string 1 corr) word) 'query))
           (corr (jit-spell--apply-correction ov corr)))))))
 
+(defun jit-spell-remove-word (word)
+  "Remove the WORD from personal dictionary or local word lists.
+Interactively offer the current word or sub-words."
+  (interactive (list (let ((cand (current-word)))
+                       (completing-read
+                        "Remove word: "
+                        (append `(,cand) (string-split cand (rx (not alpha))))
+                        nil nil nil nil cand))))
+  (let ((places nil))
+    ;; Check among the local words.
+    (save-excursion
+      (goto-char (point-min))
+      (when (re-search-forward (rx bol (syntax comment-start) (*? nonl)
+                                   (literal ispell-words-keyword) (*? nonl)
+                                   (group-n 1 (and " " (literal word))))
+                               nil t)
+        (push "the buffer's word list" places)
+        (delete-region (match-beginning 1) (match-end 1))))
+    ;; Remove it from the local word list.
+    (when (member word jit-spell--local-words)
+      (cl-callf2 delete word jit-spell--local-words)
+      (unless places (push "the session's word list" places)))
+    ;; Remove it from the personal dictionary
+    (let ((case-fold-search nil))
+      (with-temp-buffer
+        (insert-file-contents ispell-personal-dictionary nil nil nil t)
```

Comment:
I don't customize this variable, and apparently it's always nil to me. It seems you are assuming here that the variable is set to the spellchecker's default value in that case?

Reply:
That is a wrinkle, I had to set that variable for something long ago and forgot that it by default nil and more or less expected to be nil. I will try to figure out what is a reliable way of finding the default location and make changes accordingly.

Reply:
Getting the location of a personal dictionary doesn't seem straightforward. See e.g. [this spell-fu issue](https://codeberg.org/ideasman42/emacs-spell-fu/issues/4).

For aspell the default location seems to be `~/.aspell.<lang>.pws`. I don't know where other programs store the personal word lists.

Reply:
Okay, this depends too much on the spellchecker program and user configuration, so the only option is not to try to do anything about it.

Do you think the command is still meaningful even with this limitation? If so, please just add a test so that `insert-file-contents` doesn't fail if the var is nil / file doesn't exist.

(By the way, have you signed the FSF copyright assignment paperwork?)

Reply:
> Okay, this depends too much on the spellchecker program and user configuration, so the only option is not to try to do anything about it.
> 
> Do you think the command is still meaningful even with this limitation? If so, please just add a test so that `insert-file-contents` doesn't fail if the var is nil / file doesn't exist.
> 

It is meaningful at least for me and I think setting finding the location of personal dictionary and setting `ispell-personal-dictionary` is doable. I added the limitation to the documentation though maybe we can also emit a warning once per session if the command is called without `ispell-personal-dictionary` being set.

Another limitation that I have realized is that the format of personal dictionaries is not standardized either. It seems like hunspell allows for affixes along with the words. For simple words all of `ispell`, `aspell` and `hunspell` all use one line per word. So the command will work for removing words that were added by jit-spell since `ispell -a` doesn't allow for affixes as far as I can tell.

> (By the way, have you signed the FSF copyright assignment paperwork?)

Yes, I have completed the assignment and Eli confirmed that it is on file a [while back](https://lists.gnu.org/archive/html/bug-gnu-emacs/2023-05/msg00751.html)